src/SideBarControls.tsx

```diff
@@ -0,0 +1,46 @@
+import ButtonIcon from './ButtonIcon';
+import { PanelLeftClose, TextSearch, X as CloseX } from 'lucide-react';
+import { useIsMobileLayout } from './theme/useIsMobileLayout';
+
+interface SideBarControlsProps {
+  isDisabled: boolean;
+  onCloseClick: () => void;
+  onOpenSearchModal: () => void;
+}
+
+export const SideBarControls = ({
+  isDisabled,
+  onCloseClick,
+  onOpenSearchModal,
+}: SideBarControlsProps) => {
+  const isMobileLayout = useIsMobileLayout();
+  const CloseIcon = isMobileLayout ? CloseX : PanelLeftClose;
+
+  const closeIconLabel = isMobileLayout
+    ? 'Close Mobile Menu'
+    : 'Close Desktop Menu';
+
+  return (
+    <>
+      <ButtonIcon
+        disabled={isDisabled}
+        onClick={onOpenSearchModal}
+        icon={TextSearch}
+        tooltip={{
+          text: 'Search History',
+          position: 'bottom',
+        }}
+        aria-label="Search History"
+      />
+      <ButtonIcon
+        icon={CloseIcon}
+        tooltip={{
+          text: 'Close Menu',
+          position: 'bottom',
+        }}
+        aria-label={closeIconLabel}
+        onClick={onCloseClick}
+      />
+    </>
+  );
+};
```

Comment:
The only difference between the desktop and mobile versions of these components is which icon is used to close

https://github.com/user-attachments/assets/99a3ebea-d475-44dc-ab8d-e97be7527981